standardization/src/test/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter_FractionalSuite.scala

```diff
@@ -207,4 +214,169 @@ class StandardizationInterpreter_FractionalSuite extends FunSuite with SparkTest
+        .build())
+    ))
+
+    val std = StandardizationInterpreter.standardize(src, desiredSchemaWithAlters, "").cache()
+    logDataFrameContent(std)
+
+    val exp = List(
+      ("01-Positive", "+3", 3.0F, Some(3.0D), Some(3.0F), 3.0D, Seq.empty),
+      ("02-Negative", "~8123,4", -8123.4F, Some(-8123.4D), Some(-8123.4F), -8123.4D, Seq.empty),
+      ("03-Null", null, 0F, None, None, 0D, Array.fill(2)(ErrorMessage.stdNullErr("src")).toList),
+      ("04-Big", "7899012345678901234567890123456789012346789,123456789", 0F, Some(7.899012345678901E42D), Some(Float.PositiveInfinity), 7.899012345678901E42,
+        err("7899012345678901234567890123456789012346789,123456789", 1)
+      ),
+      ("05-Big II", "+1E40", 0F, Some(1.0E40D), Some(Float.PositiveInfinity), 1.0E40D, err("+1E40", 1)),
+      ("06-Big III", "2E308", 0F, Some(1000.001D), Some(Float.PositiveInfinity), Double.PositiveInfinity, err("2E308", 2)),
+      ("07-Small", "~7899012345678901234567890123456789012346789,123456789", 0F, Some(-7.899012345678901E42D), Some(Float.NegativeInfinity), -7.899012345678901E42,
+        err("~7899012345678901234567890123456789012346789,123456789", 1)
+      ),
+      ("08-Small II", "~1,1E40", 0F, Some(-1.1E40D), Some(Float.NegativeInfinity), -1.1E40D, err("~1,1E40", 1)),
+      ("09-Small III", "~3E308", 0F, Some(1000.001D), Some(Float.NegativeInfinity), Double.NegativeInfinity, err("~3E308", 2)),
+      ("10-Wrong", "hello", 0F, Some(1000.001D), Some(-1000000.0F), 0D, err("hello", 4)),
+      ("11-Infinity", "+∞", 0F, Some(1000.001D), Some(Float.PositiveInfinity), Double.PositiveInfinity, err("+∞", 2)),
+      ("12-Negative Infinity", "~∞", 0F, Some(1000.001D), Some(Float.NegativeInfinity), Double.NegativeInfinity, err("~∞", 2)),
+      ("13-Old decimal", "5.5", 0F, Some(1000.001D), Some(-1000000.0F), 0D, err("5.5", 4)),
+      ("14-Old minus", "-10", 0F, Some(1000.001D), Some(-1000000.0F), 0D, err("-10", 4)),
+      ("15-Infinity as word", "Infinity", 0F, Some(1000.001D), Some(-1000000.0F), 0D, err("Infinity", 4))
+    )
+    assertResult(exp)(std.as[(String, String, Float, Option[Double], Option[Float], Double, Seq[ErrorMessage])].collect().toList)
+  }
+
+  test("Using patterns") {
+    val input = Seq(
+      ("01-Positive", "+3°"),
+      ("02-Negative", "(8 123,4°)"),
+      ("03-Null", null),
+      ("04-Big", "+789 9012 345 678 901 234 567 890 123 456 789 012 346 789,123456789°"),
+      ("05-Big II", "+1E40°"),
+      ("06-Big III", "+2E308°"),
+      ("07-Small", "(789 9012 345 678 901 234 567 890 123 456 789 012 346 789,123456789°)"),
+      ("08-Small II", "(1,1E40°)"),
+      ("09-Small III", "(3E308°)"),
+      ("10-Wrong", "hello"),
+      ("11-Not adhering to pattern", "(1 234,56)"),
+      ("12-Not adhering to pattern II","+1,234.56°"),
+      ("13-Infinity", "+∞°"),
+      ("14-Negative Infinity", "(∞°)")
+    )
+
+    val src = input.toDF("description", "src")
+
+    val pattern = "+#,000.#°;(#,000.#°)"
```

Comment:
Apostrophe (`'`) is an escape symbol. But the "DecimalParser` is not able to parse numbers with some string in middle of them (at least my trials let me believe so).
Your example could be solved by defining the _grouping separator_ to be `#`, but it would also require to change the symbol for non-obligatory digit (which is possible, but not via `Metadata` right now).